### ISSUE TITLE
RangeError: Invalid or out-of-range index

### ISSUE THREAD

**Issue by TravisThomp** [huggingface/gsplat.js#50]
About 10% of the time when loading a scene that auto rotates a splat it will give this error and cause the splat not to load. If the error is not produced within 5 seconds of the scene loading it will not occur throughout the scene's lifetime. I'm willing to go in and try and fix this, but some guidance on why this could be happening would be useful :p

```
RangeError: invalid or out-of-range index [e4134a98-1814-4da0-be00-5b2f99d2d1b3:76:26](blob:http://localhost:5173/e4134a98-1814-4da0-be00-5b2f99d2d1b3)
    runSort blob:http://localhost:5173/e4134a98-1814-4da0-be00-5b2f99d2d1b3:76
    throttledSort blob:http://localhost:5173/e4134a98-1814-4da0-be00-5b2f99d2d1b3:101
    throttledSort blob:http://localhost:5173/e4134a98-1814-4da0-be00-5b2f99d2d1b3:105
```
The trace points to the line 
`wasmModule.HEAPF32.set(sortData.positions, positionsPtr / 4);`

from the full function 
```
  const runSort = () => {
      wasmModule.HEAPF32.set(sortData.positions, positionsPtr / 4);
      wasmModule.HEAPF32.set(sortData.transforms, transformsPtr / 4);
      wasmModule.HEAPU32.set(sortData.transformIndices, transformIndicesPtr / 4);
      wasmModule.HEAPF32.set(viewProj, viewProjPtr / 4);
      wasmModule._sort(viewProjPtr, transformsPtr, transformIndicesPtr, sortData.vertexCount, positionsPtr, chunksPtr, depthBufferPtr, depthIndexPtr, startsPtr, countsPtr);
      const depthIndex = new Uint32Array(wasmModule.HEAPU32.buffer, depthIndexPtr, sortData.vertexCount);
      const detachedDepthIndex = new Uint32Array(depthIndex.slice().buffer);
      const chunks = new Uint8Array(wasmModule.HEAPU8.buffer, chunksPtr, sortData.vertexCount);
      const detachedChunks = new Uint8Array(chunks.slice().buffer);
      self.postMessage({ depthIndex: detachedDepthIndex, chunks: detachedChunks }, [
          detachedDepthIndex.buffer,
          detachedChunks.buffer,
      ]);
  };
```
I'm assuming it occurs from a memory issue when loading the the rotated splat, do to some weird fault. 

The code to reproduce it (This is for Svelte currently but can easily be dragged out if not using)
```
<script lang="ts">
  import { onMount } from "svelte";
  import * as SPLAT from "gsplat";
  import { Splat, Camera, Scene } from "gsplat";

  function loadModel(scene: Scene): Promise<Splat> {
    const modelURL = "https://huggingface.co/datasets/dylanebert/3dgs/resolve/main/bonsai/bonsai-7k.splat";
    let head = SPLAT.Loader.LoadAsync(modelURL, scene, () => {});

    return head;
  }

  function updateModelRotation(model: Splat, tick: number) {
    const autoRotateMultipler = 0.0025;

    // ERROR CAUSED BY THIS LINE(atleast it doesnt show when commented out)
    model.rotation = SPLAT.Quaternion.FromEuler(
      new SPLAT.Vector3(0, tick * autoRotateMultipler, 0)
    );
  }

  onMount(async () => {
    const canvas = document.getElementById("canvas") as HTMLCanvasElement;

    const renderer = new SPLAT.WebGLRenderer(canvas, null);
    canvas.style.background = "transparent";

    const scene = new SPLAT.Scene();
    const camera = new SPLAT.Camera();

    let model = await loadModel(scene);

    let tick: number = 0;
    const frame = () => {
      updateModelRotation(model, tick);

      tick += 1;
      renderer.render(scene, camera);
      requestAnimationFrame(frame);
    };

    const handleResize = () => {
      renderer.setSize(canvas.clientWidth, canvas.clientHeight);
    };
    requestAnimationFrame(frame);

    window.addEventListener("resize", handleResize);
  });
</script>
```

**Comment by TravisThomp** [comment]
The reproduction of this can be made more successful (up to 100%) if you load in multiple splats then constantly rotate at least one of them.

**Comment by dylanebert** [comment]
Thanks for pointing this out, this may be related to the flickering issue #48 

Now it's almost certainly coming from an index alignment bug in the sorting. I'll look into it this week as well

**Comment by TravisThomp** [comment]
I was looking into it a little bit today, it seemed to me that it was throwing from the runSort function inside the SortWorker.ts class. One of the issues I was running into was when copying the positions data into wasm (when a lot of gaussians are loaded) `wasmModule.HEAPF32.set(sortData.positions, positionsPtr / 4);` can cause an overflow which will always throw the Range Error. So increasing the memory of the HEAP32F in wasm may help(or at least adding a check and warning the user). I compressed my gaussians to try and get around this but I still run into this range error for another reason, haven't had time to really look into that but I'm assuming it's related to the index alignment bug you mentioned. Also just for more information, when the range error is thrown with the rotation of models, sometimes nothing will load, other times it rotate 180 then rotate back the 180 as a distorted version of the original splat. Let me know if I can do anything to help :p

**Comment by dylanebert** [comment]
I believe I've fixed this issue in #53, which solves concurrency issues in `SortWorker.ts`

Please confirm and reopen if you still encounter issues!
